### DISCUSSION TITLE
Fast API Backend

### DISCUSSION THREAD

**Discussion by gabrielolympie** [Ironclad/rivet#287] [General]
Edit: Link to the current experimental project: https://github.com/gabrielolympie/VisualAIFlows/tree/main

I started building a fast api backend for rivet AI, to be able to leverage this excellent visual tool with python functions and methods and i am thinking to open source it,

Would some of you be interested to try it and/or contribute (i added helpers so including a new component is as simple as implementing it and adding a new route to fast api).

Functionality so far on top of rivet:

- presets : dictionnary that you can use to store your presets and use them easily from the interface

- methods: execution of a python method through an http request to fastapi

- flows : complex flows including handling loop, tools etc. (i have in mind this framework in the future https://github.com/epfl-dlab/aiflows)

Plus the creation of the playground is automated, and load all the components on the canvas to be used easily.

![image](https://github.com/Ironclad/rivet/assets/46057918/453058d2-7772-4a15-aecd-163d507e53cd)
![image](https://github.com/Ironclad/rivet/assets/46057918/353a35b9-6fdb-420d-9891-7be1e0a3719a)
![image](https://github.com/Ironclad/rivet/assets/46057918/92d6c582-c6ed-4bc6-9b49-b0f186d25369)

**Comment by abrenneke** [comment]
This looks really interesting! I'd love to hear more!

**Reply by gabrielolympie** [reply]
There you go, don't hesitate to provide feedback if you find some improvements area :)

https://github.com/gabrielolympie/VisualAIFlows/tree/main

**Reply by gabrielolympie** [reply]
btw, just saw you are actually one of the top contributor of rivet and just wanted to say that how it is designed is brillant. I actually was on a spree tying to implement a python visual flow editor myself similar to ryven project, but was strugling to design the front and the actual orchestration of cyclic directed flow graph, and the way you managed to handle it through branching and for loop orchestrator is just simple but very elegant solution, very fan of the project! The only part missing was a sdk enabling to conect it with python community ;)
